addons/tts-subtitles/userscript.js

```diff
@@ -0,0 +1,41 @@
+export default async function ({ addon, console }) {
+  let subtitleBox;
+  const createSubtitleBox = async () => {
+    if (addon.tab.editorMode !== "projectpage" && addon.tab.editorMode !== "editor") {
+      return null;
+    }
+    const player =
+      addon.tab.editorMode === "projectpage"
+        ? (await addon.tab.waitForElement(".guiPlayer")).parentElement
+        : await addon.tab.waitForElement("[class^='stage-wrapper_stage-wrapper']");
+    const box = document.createElement("div");
+    box.classList.add("sa-tts-subtitles-box");
+    player.insertAdjacentElement("afterend", box);
+    return box;
+  };
+  addon.tab.addEventListener("urlChange", async () => {
+    subtitleBox = await createSubtitleBox();
+  });
+  subtitleBox = await createSubtitleBox();
+
+  const addToSubtitlesBox = (text) => {
+    if (!subtitleBox) {
+      return;
+    }
+    subtitleBox.textContent += text + "\n";
+    subtitleBox.scrollTop = subtitleBox.scrollHeight;
+  };
+
+  // It needs to be on window.Promise for some reason
+  // Otherwise, there's an error that when googled yields zero results
```

Comment:
Hehe, welcome to the world of traps and polluting 😅 

It's likely you forgot to use "bind". It's also possible you messed up with the promises. I don't think it's a good idea to use await inside a function that returns a promise. I might change this code a bit before merging

Reply:
I don't think I used `bind` - I'll see if that fixes things.

Reply:
It doesn't like the `this` value.

```js
  // Uncaught TypeError: [object Module] is not a constructor
  const oldRace = Promise.race;
  window.Promise.race = (args) => {
    const result = oldRace.bind(this)(args);
```

Reply:
Arrow functions do not have a defined `this` value. Try a normal function.

Reply:
> const result = oldRace.bind(Promise)(args);

Well, I guess that works too. But now you know why "this" wasn't working.